lattices/src/lib.rs

```diff
@@ -107,3 +107,22 @@ pub trait IsTop {
     /// Returns if `self` is lattice top (⊤).
     fn is_top(&self) -> bool;
 }
+
+/// Trait to atomize a lattice into individual elements. For example, a [`set_union::SetUnion`]
+/// will be broken up into individual singleton elements.
+///
+/// Formally, breaks up `Self` into an set of lattice points forming a (strong) [antichain](https://en.wikipedia.org/wiki/Antichain).
+pub trait Atomize: Merge<Self::Atom> {
+    /// The type of atoms for this lattice.
+    type Atom: 'static;
+
+    /// The iter type iterating the antichain atoms.
+    type AtomIter: 'static + Iterator<Item = Self::Atom>;
+
+    /// Atomize self: convert into an iter of atoms.
+    ///
+    /// Must always return at least one value.
```

Comment:
I think I understand now why it needs to produce always at least one value. Because without it then you can't re-form lattices without a bot. For example Min and Max. I guess you can work around that with the reduce trick where you take the first item and then merge subsequent atoms into that one, and you would panic if the iterator produces zero items but maybe you're escaping the type system then.

Reply:
It allows composition to work. If we have a `MapUnion<K, SetUnion<T>>`:

Atomize
```
{
my_key: {},
}
```
Will atomize as `(my_key, {})`

Instead of just atomizing to empty, in which case we lose the key